install.ps1

```diff
@@ -1,429 +1,888 @@
+        $newXmlNode.SetAttribute("name", "VM_COMMON_DIR")
+        $newXmlNode.SetAttribute("value", $vmCommonDirText.text);
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"))
+        $newXmlNode.SetAttribute("name", "TOOL_LIST_DIR")
+        $newXmlNode.SetAttribute("value", $toolListDirText.text);
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"))
+        $newXmlNode.SetAttribute("name", "TOOL_LIST_SHORTCUT")
+        $newXmlNode.SetAttribute("value", $toolListShortCutText.text);
+        $newXmlNode = $envs.AppendChild($configXml.CreateElement("env"))
+        $newXmlNode.SetAttribute("name", "RAW_TOOLS_DIR")
+        $newXmlNode.SetAttribute("value", $rawToolsDirText.text)
+
+        # Add selected packages
+        $packages = $configXml.SelectSingleNode('//packages')
+        foreach($package in $selectedPackagesBox.Items) {
+            $newXmlNode = $packages.AppendChild($configXml.CreateElement("package"))
+            $newXmlNode.SetAttribute("name", $package)
+        }
     } else {
-      Install-BoxStarterPackage -PackageName "flarevm.installer.flare" -Credential $cred
+        Write-Host "[+] Cancel pressed, stopping installation..."
+        Start-Sleep 3
+        exit 1
     }
-  }
-  exit 0
+
+    ################################################################################
+    ## END GUI
+    ################################################################################
 }
 
-# The necessary basic environment variables
-$EnvVars = @(
-  "VM_COMMON_DIR",
-  "TOOL_LIST_DIR",
-  "TOOL_LIST_SHORTCUT",
-  "RAW_TOOLS_DIR"
-  )
-
-foreach ($envVar in $EnvVars) {
-  try {
-    $value = [Environment]::ExpandEnvironmentVariables($loaded_profile.env.($envVar))
-    if (-Not (Set-EnvironmentVariableWrap $envVar $value)) {
-      Write-Warning "[-] Failed to set environment variable $envVar"
-    }
-  } catch {}
+# Save the config file
+Write-Host "[+] Saving configuration file..."
+$configXml.save($configPath)
+
+# Parse config and set initial environment variables
+Write-Host "[+] Parsing configuration file..."
+foreach ($env in $configXml.config.envs.env) {
+    $path = [Environment]::ExpandEnvironmentVariables($($env.value))
+    Write-Host "`t[+] Setting %$($env.name)% to: $path" -ForegroundColor Green
+    [Environment]::SetEnvironmentVariable("$($env.name)", $path, "Machine")
 }
+refreshenv
 
-choco install -y common.fireeye
+# Install the common module
+# This creates all necessary folders based on custom environment variables
+Write-Host "[+] Installing shared module..."
+choco install common.vm -y --force
 refreshenv
+$configXml.save((Join-Path ${Env:VM_COMMON_DIR} "config.xml"))
+
+# Log basic system information to assist with troubleshooting
+Write-Host "[+] Logging basic system information to assist with any future troubleshooting..."
+Import-Module "${Env:VM_COMMON_DIR}\vm.common\vm.common.psm1" -Force -DisableNameChecking
+VM-Get-Host-Info
```

Comment:
This appears to appear twice in the log I get towards the end of the install?

Reply:
Also these appear empty (no values below)?
```
VM Boxstarter Version
-----


VM Installed Packages
-----
```

Reply:
https://github.com/mandiant/VM-Packages/pull/119 fixes the missing / empty values, good find!

Reply:
I don't think I ran this twice. Testing again now with the updates.